File: week04/TimeWatch_java/TimeWatch.java
import java.lang.*;

public class TimeWatch {
    public static void main(String[] args) {
        long start_time = System.nanoTime();
        boolean state = new AlgorithmOne().run();
        if (state) System.out.println("All OK.");
        else System.out.println("Something went wrong!");
        long end_time = System.nanoTime();
        System.out.println("Time: "+ (end_time - start_time)/1000000+ " ms");
    }
}

File: week04/TimeWatch_c++/TimeWatch.cpp
#include "AlgorithmOne.h"
#include <iostream>

int main() {
  	std::clock_t start = std::clock();
	auto t_start = std::chrono::high_resolution_clock::now();
    bool state = (new AlgorithmOne())->run();
    if (state) std::cout << "All OK.\n";
    else std::cout << "Something went wrong!\n";
    std::clock_t end = std::clock();
	auto t_end = std::chrono::high_resolution_clock::now();
   std::cout<< "Time: " << 1000.0*(end - start)/CLOCKS_PER_SEC<< " -- " 
   << std::chrono::duration<double, std::milli>(t_end-t_start).count() << " ms "<< std::endl;
    return 0;
 }